qiskit/circuit/quantumcircuit.py

```diff
@@ -1317,19 +1397,7 @@ def bind_parameters(self, value_dict):
         Returns:
             QuantumCircuit: copy of self with assignment substitution.
         """
-        new_circuit = self.copy()
-        unrolled_value_dict = self._unroll_param_dict(value_dict)
-
-        if unrolled_value_dict.keys() > self.parameters:
-            raise CircuitError('Cannot bind parameters ({}) not present in the circuit.'.format(
-                [str(p) for p in value_dict.keys() - self.parameters]))
-
-        for parameter, value in unrolled_value_dict.items():
-            new_circuit._bind_parameter(parameter, value)
-        # clear evaluated expressions
-        for parameter in unrolled_value_dict:
-            del new_circuit._parameter_table[parameter]
-        return new_circuit
+        return self.assign_parameters(value_dict, inplace=False)
```

Comment:
If we are going to keep `bind_parameters`, it should keep the behavior that it raises if passed a non-float.